libs/application/templates/children-residence-change/src/fields/Overview/SignatureModal/ErrorMessage.tsx

```diff
@@ -1,21 +1,41 @@
-import React from 'react'
-import { useIntl } from 'react-intl'
+import React, { useState, useEffect } from 'react'
```

Comment:
Just a suggestion, different style really and without useEffect or useState. I've not tested this implementation though.
```
import React from 'react'
import { useIntl } from 'react-intl'
import { Box, AlertMessage, Button } from '@island.is/island-ui/core'
import { signatureModal } from '../../../lib/messages'

const errorMessages = {
  404: signatureModal.error.noElectronicId,
  400: signatureModal.error.userCancelled,
  409: signatureModal.error.timeOut,
}

interface ErrorMessageProps {
  onClose: () => void
  errorCode?: keyof typeof errorMessages
}

const ErrorMessage = ({ onClose, errorCode }: ErrorMessageProps) => {
  const { formatMessage } = useIntl()
  return (
    <>
      <AlertMessage
        type="error"
        title={formatMessage(signatureModal.defaultError.title)}
        message={formatMessage(
          errorCode
            ? errorMessages[errorCode]
            : signatureModal.defaultError.message,
        )}
      />
      <Box marginTop={3} justifyContent="center">
        <Button onClick={onClose} variant="primary">
          {formatMessage(signatureModal.general.closeButtonLabel)}
        </Button>
      </Box>
    </>
  )
}

export default ErrorMessage
```

Reply:
This didn't quite work but did a similar approach.